cpp/zeder_to_zotero_importer.cc

```diff
@@ -0,0 +1,432 @@
+
+
+namespace {
+
+
+using namespace ZoteroHarvester;
+
+
+[[noreturn]] void Usage() {
+    std::cerr << "Usage: " << ::progname << " [options] config_file_path mode zeder_flavour zeder_ids fields_to_update\n"
+              << "\n"
+              << "\tOptions:\n"
+              << "\t[--min-log-level=log_level]     Possible log levels are ERROR, WARNING (default), INFO and DEBUG\n"
+              << "\t[--overwrite-on-import]         Overwrite existing entries/sections when performing an import\n"
+              << "\n"
+              << "\tconfig_file_path                Path to the Zotero Harvester config file\n"
+              << "\tmode                            Either IMPORT or UPDATE\n"
+              << "\tzeder_flavour                   Either IXTHEO or KRIMDOK\n"
+              << "\tzeder_ids                       Comma-separated list of Zeder entry IDs to import/update.\n"
+              << "\t                                Special-case for updating: Use '*' to update all entries found in the config that belong to the Zeder flavour\n"
+              << "\tfields_to_update                Comma-separated list of the following fields to update: \n"
+              << "\t                                \tONLINE_PPN, PRINT_PPN, ONLINE_ISSN, PRINT_ISSN, EXPECTED_LANGUAGES, ENTRY_POINT_URL, UPLOAD_OPERATION, UPDATE_WINDOW.\n"
+              << "\t                                Ignored when importing entries (all importable fields will be imported).\n\n";
+    std::exit(EXIT_FAILURE);
+}
+
+
+struct CommandLineArgs {
+    enum Mode { INVALID, IMPORT, UPDATE };
+
+    bool overwrite_on_import_;
+    std::string config_path_;
+    Mode mode_;
+    Zeder::Flavour zeder_flavour_;
+    std::set<unsigned> zeder_ids_;
+    std::set<Config::JournalParams::IniKey> fields_to_update_;
+public:
+    explicit CommandLineArgs() : overwrite_on_import_(false), mode_(INVALID) {}
+};
+
+
+void ParseCommandLineArgs(int * const argc, char *** const argv, CommandLineArgs * const commandline_args) {
+    while (StringUtil::StartsWith((*argv)[1], "--")) {
+        if (std::strcmp((*argv)[1], "--overwrite-on-import") == 0) {
+            commandline_args->overwrite_on_import_ = true;
+            --*argc, ++*argv;
+            continue;
+        }
+    }
+
+    if (*argc < 5)
+        Usage();
+
+    commandline_args->config_path_ = (*argv)[1];
+    --*argc, ++*argv;
+
+    const std::string mode((*argv)[1]);
+    --*argc, ++*argv;
+
+    if (::strcasecmp(mode.c_str(), "IMPORT") == 0)
+        commandline_args->mode_ = CommandLineArgs::Mode::IMPORT;
+    else if (::strcasecmp(mode.c_str(), "UPDATE") == 0)
+        commandline_args->mode_ = CommandLineArgs::Mode::UPDATE;
+    else
+        Usage();
+
+    const std::string zeder_flavour((*argv)[1]);
+    --*argc, ++*argv;
+
+    commandline_args->zeder_flavour_ = Zeder::ParseFlavour(zeder_flavour);
+
+    const std::string zeder_id_list((*argv)[1]);
+    --*argc, ++*argv;
+
+    std::set<std::string> buffer;
```

Comment:
Would zeder_ids be a better name than buffer?

Reply:
It's used for multiple argument types, therefore went with a generic name.